demo/ios/BackgroundService/PorcupineBackgroundServiceDemo/ViewController.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Copyright 2018-2021 Picovoice Inc.
+//  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+//  file accompanying this source.
+//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+//  specific language governing permissions and limitations under the License.
+//
+
+import UIKit
+import Porcupine
+
+class ViewController: UIViewController, UITextViewDelegate {
+    
+    @IBOutlet weak var startButton: UIButton!
+    @IBOutlet weak var textView: UITextView!
+    
+    var wakeWord = Porcupine.BuiltInKeyword.porcupine
+
+    var porcupineManager: PorcupineManager!
+    var isRecording = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        startButton.layer.cornerRadius = 0.5 * startButton.bounds.size.width
+        startButton.clipsToBounds = true
+        
+        textView.text = "Press the Start button and say the wake word \"Porcupine\". You may close the app but it must be running in the background."
+    }
+
+    @IBAction func toggleStartButton(_ sender: UIButton) {
+        if !isRecording {
+            NotificationManager.shared.requestNotificationAuthorization()
+            
+            let keywordCallback: ((Int32) -> Void) = { keywordIndex in
+                NotificationManager.shared.sendNotification()
+            }
+
+            do {
+                porcupineManager = try PorcupineManager(keyword: wakeWord, onDetection: keywordCallback)
```

Comment:
you can probably initialize PorcupineManager once when the app is launched and then just start/stop the manager in this function.